handlers/getakt.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Module for requesting documents via AktDirekt, Lantmäteriet
+ * Usage: 
+ * Check for service health, response "UP" or "DOWN" - /origoserver/healthcheck
+ * Get a djvu index page that gets all pages included in akt(document) - /origoserver/document/index.djvu?archive={archive/county-number}&id={akt-number}
+ * Get a single djvu page - /origoserver/document/page_{vers}_{subdoc}_{page}_{archive}_{id}.djvu
+ * 
+ */
+
+// Dependencies
+const url = require('url');
+const http = require('http');
+const httpProxy = require('http-proxy');
+const config = require('../conf/config');
+const getToken = require('../lib/tokenrequest');
+
+// Use the config for AktDirekt
+const akturl = config.getAkt.url;
+const aktkey = config.getAkt.consumer_key;
+const aktsecret = config.getAkt.consumer_secret;
+const aktscope = config.getAkt.scope;
+
+// Token holder
+let token;
+
+// Proxy server holder
+let proxy = httpProxy.createProxyServer();
+
+// Get hold of that nested token and use in proxy request header
+function tokenHeader() { };
+
+// Proxy request config
+proxy.on('proxyReq', (proxyReq, req) => {
+
+  // Get the request path
+  const parsedUrl = url.parse(req.url);
+  const query = parsedUrl.search;
+  const path = parsedUrl.pathname;
+  const pathPart = path.split("_");
+
+  const pathObj = {
+    page_: pathPart[0],
+    vers: pathPart[1],
+    subdoc: pathPart[2],
+    page: pathPart[3],
+    archive: pathPart[4],
+    enc_id: pathPart[5]
+  }
+
+  // Proxy path holder
+  let proxyPath;
+
+  if (path.includes('healthcheck')) {
+    const healthcheck = `/distribution/produkter/aktdirekt/v3.0${path}`;
+    proxyPath = healthcheck;
+  } else if (path.includes('index.djvu')) {
+    const index = `/distribution/produkter/aktdirekt/v3.0${path}${encodeURI(query)}`;
+    proxyPath = index;
+  } else if (path.includes('page_')) {
+    const page = `/distribution/produkter/aktdirekt/v3.0${pathObj.page_}_${pathObj.vers}_${pathObj.subdoc}_${pathObj.page}_${pathObj.archive}_${encodeURIComponent(pathObj.enc_id)}`;
+    proxyPath = page;
+  }
+  // Set proper proxy request path
+  proxyReq.path = proxyPath;
+  // Set the proxy request headers
+  proxyReq.setHeader(`Authorization`, `Bearer ${token}` + ` scope`, `${aktscope}`);
+})
+
+// If the http-proxy throws an error, log it
+proxy.on('error', (err) => {
+  console.log('AktDirekt proxy error: ', err);
+})
+
+// Do the request in proper order
+const getAkt = async (req, res) => {
+  // Call the promised token
+  await getToken(aktkey, aktsecret, aktscope)
+    .then(JSON.parse)
+    .then((result) => {
+      token = result.access_token;
+      tokenHeader(token);
+    })
+    .catch((e) => {
+      console.log(e);
+    })
+  // Send the request
+  await proxy.web(req, res, {
+    target: akturl,
+    secure: false,
```

Comment:
Maybe the secure parameter could be an config option?

Reply:
I am not sure that would do any for the user. Any special scenarios you think about?

Reply:
The user might want to configure web proxy params for example secure and could do so in the config file with a web option in the getAkt config.  Web proxy settings are related with comsuer key, secret and scope. imo. Default settings could be as is but have an option to be overridden from the config.

Reply:
I am running origo-server on IIS with SSL cert and secure option set to false. If secure option is set to false "only" means that it will not attemp verify the SSL cert. I cant run the proxy in secure mode as it is. 
How are you running origo-server with SSL? Have you manage to ru it in secure mode?
If there is a need to run it in secure mode I guess we need to add more proxysettings to the service config. 
[https://github.com/nodejitsu/node-http-proxy#https---http](url)

Reply:
The origo-server solution in Västerås uses the IIS redirect to pipe requests to Origo server. Those requests are not encrypted (but maybe they should be). I will resolve this conversation since this functionallity may not be needed right now.